File: HomeworkManager/HomeworkManager.Model/CustomEntities/Assignment/UpdatedAssignment.cs
using HomeworkManager.Model.Constants;

namespace HomeworkManager.Model.CustomEntities.Assignment;

public class UpdatedAssignment
{
    public required int AssignmentId { get; set; }
    public required string Name { get; set; }
    public required string? Description { get; set; }
    public required DateTime Deadline { get; set; }
    public required AssignmentTypeId AssignmentTypeId { get; set; }
    public required bool PresentationRequired { get; set; }
}

File: HomeworkManager.Angular/ClientApp/src/app/shared-module/models/assignment/updated-assignment.ts
import { AssignmentTypeId } from "./assignment-type-id";

export class UpdatedAssignment {
  assignmentId!: number;
  name!: string;
  description!: string;
  deadline!: string;
  presentationRequired!: boolean;
  assignmentTypeId!: AssignmentTypeId;
}